fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/StringExpNodeVisitor.java

```diff
@@ -0,0 +1,406 @@
+        result.append(" - ");
+        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String mult(String left, String right) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" * ");
+        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String div(String left, String right) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" / ");
+        result.append(right);
+        return result.toString();
+    }
+
+    @Override
+    public String isNull(String expr) {
+        StringBuilder result = new StringBuilder();
+        result.append(expr);
+        result.append(" IS NULL");
+        return result.toString();
+    }
+
+    @Override
+    public String isNotNull(String expr) {
+        StringBuilder result = new StringBuilder();
+        result.append(expr);
+        result.append(" IS NOT NULL");
+        return result.toString();
+    }
+
+    @Override
+    public String between(String leftValue, String rightValue) {
+        StringBuilder result = new StringBuilder();
+        result.append(" BETWEEN ");
+        result.append(leftValue);
+        result.append(" AND ");
+        result.append(rightValue);
+        return result.toString();
+    }
+
+    @Override
+    public String column(String tableAlias, String columnName) {
+        StringBuilder result = new StringBuilder();
+        if (tableAlias != null) {
+            result.append(tableAlias);
+            result.append(".");
+        }
+        result.append(columnName);
+        return result.toString();
+    }
+
+    @Override
+    public String bindMarker(Double value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new DoubleBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Long value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new LongBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(String value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new StringBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Instant value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new InstantBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String bindMarker(Integer value) {
+        if (this.bindMarkers != null) {
+            bindMarkers.add(new IntegerBindMarkerNode(value));
+        }
+        return "?";
+    }
+
+    @Override
+    public String in(String left, List<String> args) {
+        StringBuilder result = new StringBuilder();
+        result.append(left);
+        result.append(" IN (");
+        result.append(String.join(",", args));
```

Comment:
encode args?

Reply:
Values are already encoded at this point:
```
    public void inLiteral(String[] inList) {
        // Parameters are encoded for security, but bind markers are always preferred and
        // recommended for security.
        List<ExpNode> args = Arrays.asList(inList).stream().map(v -> new StringExpNode(encodeParameter(v))).collect(Collectors.toList());
        addNode(new InListExpNode(args));
    }
```
But I'm thinking about doing encoding inside the visitor - probably safer in the end.

Reply:
Switching to encoding inside the StringExpNodeVisitor